### ISSUE TITLE
Extraordinarily slow release cycle

### ISSUE THREAD

**Issue by DrKittens** [Haptic-Apps/Slide#3449]
Slide version: 6.7.1
Android version: 12

Just posting an issue to comment that a known core feature breaking bug has been fixed, ex #3439 and no point update has occurred to push it to the userbase.

Please push a release version to f-droid and adjust whatever Metrics you consider doing point releases for to include critical feature bug fixes.

**Comment by TacoTheDank** [comment]
@DrKittens Slide has been basically in low maintenance mode for a long while because the codebase is so godawful it's pretty much impossible maintain long term. Slide has no release timeline.

That being said, I will consider doing a point release. GPlay will not be able to get it though due to technical requirements that need fixing.

**Comment by SolarAquarion** [comment]
The code basically needs to be rebased in order for it to be maintainable.

**Comment by daffydock** [comment]
If this is the case, outside small, sporadic updates/fixes, is the project considered stalled from having any major updates? It strikes me as such, however I just want to be sure. Since I have seen a couple of up-keep commits in the last couple of days. 

I was under the impression that a major update was coming at some point, in version 7. Albeit that does not seem to be in the cards, now.

**Comment by TacoTheDank** [comment]
>If this is the case, outside small, sporadic updates/fixes, is the project considered stalled from having any major updates?

For now, yes, but this could change in the future. My focus is currently on other projects and college for right now, and programming is more of a hobby thing for me anyway. I would like to come back to this project in full force at some point and overhaul it :)

So at least for now, it is not considered abandoned. If this were to be the case, it would be stated as such in the README for everyone to know.

**Comment by daffydock** [comment]
That's fair. 

Albeit I never suggested abandoned. An app I follow has not had an update in over a year, it is not considered 'abandoned' either, however within the context of that app, the unfixed security issues have added up to the point that albeit not labeled as such, it would just be unwise to use it, today.  

You stated you would be doing minor updates previously, and I was wondering if larger updates would ever come as the app ages. Your answer does address that. Thank you for the minor updates, they are welcome.

**Comment by TacoTheDank** [comment]
I'm aware abandonment wasn't brought up lol, this was just in case anyone else came across this issue and was wondering about the current state of the project :)

Some turning over of hands has occurred within the past 8 or so months, with the app's creator (ccrama) telling me that (in short) basically he is unable to spend much, if any, time on the projects due to being busy with life stuff. We had the projects transferred over to an organization for easier management.
I will say that Slide isn't necessarily a big priority for me either at the moment because of college, other projects I work on, and all sorts of other things with which I'm kept busy.

Apologies for the info dump lol, I just felt it would be good for those interested to know the projects' current status so they're not left in the dark.

**Comment by TacoTheDank** [comment]
For the time being, if you no longer want to use Slide or deal with its large plethora of issues, I would recommend these much better maintained open-source alternatives:
- https://github.com/Docile-Alligator/Infinity-For-Reddit
- https://github.com/QuantumBadger/RedReader
- https://github.com/Tunous/Dawn

**Comment by daffydock** [comment]
Thanks. As far myself, I have RR and IfR already on my phone. RR is great for added privacy and is lighting fast. IfR is what I use often and have tweaked to work a bit like Slide, since it is easier to upload pics, but I keep Slide for the widgets. Since the other 2 lack them. Its a mess, but a useful one.

**Comment by DrKittens** [comment]
I appreciate the transparency from the current maintainer. It has been nearly 2 months since you mentioned consideration for doing a point release to f-droid and 6.7.1 is still the current "release". Was the decision to not do a point release / does the current code fail to compile?

As is, I'll take the advice to churn to a different tool.

**Comment by TacoTheDank** [comment]
> Was the decision to not do a point release / does the current code fail to compile?

It's because I don't currently have the signing keys to build the app. ccrama has them, but because his activity is so spotty, I don't know when he might send me the keys :(

I COULD technically build the APK and just release it, but because the signing keys would be different, it would require that you uninstall the app and then reinstall my build (which would be quite inconvenient for most people).

**Comment by DrKittens** [comment]
Fair enough, glad to know it builds but signing keys are the issue.

Correct me if im wrong but doesnt the F-droid build just scrape github for new release dot points then f-droid itself builds the project?

ref: https://f-droid.org/en/docs/FAQ_-_App_Developers/#do-i-need-to-tell-you-when-i-update
https://f-droid.org/en/docs/Build_Metadata_Reference/#UpdateCheckMode


So if it successfully builds, wouldn't f-droid downstream a compiled version from that github release automagically?

Ex a new release to f-droid should be as simple as a new release tag in git & potentially a version bump / update in app & gradle manifests then wait a few days for f-droids build server to do its thing?

**Comment by TacoTheDank** [comment]
> Correct me if im wrong but doesnt the F-droid build just scrape github for new release dot points then f-droid itself builds the project?

This is correct. However, I would also like to attach an APK (signed by the proper keys) to the release if possible. I may just decide to make a new release regardless.

**Comment by Kommynct** [comment]
What happened to the planned rewrite?

**Comment by daffydock** [comment]
@kommynct

Slide's Dev has been on hiatus. Thus the rewrite did not happen and it is an unknown if it will ever happen.

**Comment by TacoTheDank** [comment]
I forgot to mention that I added an indefinite hiatus note to the README.
